### PR TITLE
WIP: Init typedefs

### PR DESCRIPTION
**Pull request by austinberke**
Working on solidifying the schema for GraphQL. @awolk would you be able to paste here the notes you were taking on Friday for the schema layout?

### CODE REVIEW

**Comment by awolk** [comment]
Awesome! Here are my rough notes for the data model:
```
Users (recruiters)

Forms
 - Owners: [User]
 - Questions:
 - Submissions (each have:)
   - Email
   - [Answer]
   - timestamp
   - Scoring: Map [User -> (Score, Notes)]
   - Emails Sent
 - Interview Slots:
	Slots: (Start, End, Applicant?)
```

File: server/src/handlers/graphql-server.ts [hunk 1]
```diff
@@ -1,9 +1,33 @@
...
-    time: Int
+  type User {
+    id: Int!
+    username: String
+  }
+
+  type Question {
+    id: Int!
+    prompt: String
+  }
```
[the remark this answers is not in the document]
Comment (austinberke) [reply server/src/handlers/graphql-server.ts]: Good idea, and I think this will nicely correspond to the Strategy DP in our UML

**Comment by austinberke** [comment]
Updated params to be required, put in basic query typedef to support retrieving a user by username and form (w/ constituent questions and answers) by user. PTAL, if good I'll work on corresponding resolvers.

File: server/src/handlers/graphql-server.ts [hunk 2]
```diff
@@ -2,8 +2,49 @@ import { ApolloServer, gql } from "apollo-server-koa";
...
+
+  type TextQuestion implements Question {
+    id: Int!
+    prompt: String!
+  }
+
+  type Answer {
+    id: Int!
+    question: Question!
+    answer: String!
```
[the remark this answers is not in the document]
Comment (austinberke) [reply server/src/handlers/graphql-server.ts]: I don't see a particular reason to maintain the "dimensionality" of answers when displaying for candidate review. My design suggests that text answers can be displayed as raw text, and multiple selections can be concatenated into a delimited string.

File: server/src/handlers/graphql-server.ts [hunk 3]
```diff
@@ -2,8 +2,49 @@ import { ApolloServer, gql } from "apollo-server-koa";
...
+
+  type Response {
+    id: Int!
+    respondent: String!
+    answers: [Answer]!
+  }
+
+  type Form {
+    id: Int!
+    author: User!
```
[the remark this answers is not in the document]
Comment (austinberke) [reply server/src/handlers/graphql-server.ts]: Again I think (especially with our deadline) we should be biased toward simpler implementations. Unless there's a strong argument supporting multiple authors I think we should maintain a 1-1 relationship with authors and their created forms.

[the remark this answers is not in the document]
Comment (awolk) [reply server/src/handlers/graphql-server.ts]: Sounds good, I think you still have to push your latest changes to this branch but go ahead and merge when you want

File: server/src/handlers/graphql-server.ts [hunk 4]
```diff
@@ -2,8 +2,49 @@ import { ApolloServer, gql } from "apollo-server-koa";
...
 const typeDefs = gql`
   type Query {
-    hello: String
-    time: Int
+    user(username: String!): User
+    form(user: User!): Form
+  }
+
+  type User {
+    id: Int!
```
Comment (dkb868) [inline server/src/handlers/graphql-server.ts]: id in Mongo is a String unless we're making our own custom IDs.

Comment (dkb868) [reply server/src/handlers/graphql-server.ts]: o jk this is already mentioned, guess changes just haven't been pushed yet

**Comment by dkb868** [comment]
Are you planning to add interview slots as well or are we leaving that out?

File: server/src/handlers/graphql-server.ts [hunk 5]
```diff
@@ -2,8 +2,49 @@ import { ApolloServer, gql } from "apollo-server-koa";
...
+  type Answer {
+    id: Int!
+    question: Question!
+    answer: String!
+  }
+
+  type Response {
+    id: Int!
+    respondent: String!
+    answers: [Answer]!
```
Comment (dkb868) [inline server/src/handlers/graphql-server.ts]: What about scoring?

Comment (awolk) [reply server/src/handlers/graphql-server.ts]: Good catch!

**Comment by austinberke** [comment]
Oops, was working on a different branch. Added support for scoring and scheduling. Going to merge to master and begin work on resolvers (will try to get some tangible stuff pushed by Sunday night). Let me know if you have problems with any typedefs in this commit @awolk @dkb868

**Pull request merged**